install-deps.sh

```diff
@@ -221,6 +221,26 @@ function install_boost_on_ubuntu {
 
 }
 
+function install_llvm_repo_on_ubuntu {
```

Comment:
As noted in private discussion, I think we should avoid becoming the llvm installer; future versions might well change things inside the script and we'd never know it. (Not that having the cached version is a great defense against that either, but at least there's a paper trail)

Reply:
Well, install-deps.sh already has various functions to install other packages from non-default repos - so install-deps already does that for these other things (gcc using a PPA and boost for examples). Plus I don't love that that the llvm script is external (and unversioned - I much prefer version control for my paper trails). But I really don't care enough to argue for one way or another. I can get it working using the external script and move on with things

Reply:
yes, I know we install things from non-distro repos, but I think the line of "and use standard code to do it that we don't own" is still worthwhile to enforce.  I also do not love the external script, but I like it better than reimplementing it.  Ideally, we'd be building Ceph with standard tools provided by the distros, but a whole herd of horses has completely destroyed that barn